### ISSUE TITLE
SL3-1-2E: Evaluation pipeline

### ISSUE THREAD

**Issue by phuepwint-thwe** [AIT-brainlab/SignLanguage#60]
- Integrate the SONAR text decoder to generate text directly from predicted embeddings.
- Evaluate on PHOENIX-2014-T test set using BLEU , BLEURT, and ROUGE. 
- Compare results against paper-reported numbers.

### Deliverables

- [x] Evaluation script
- [x]  test set scores per metric
- [x] comparison table

**Estimate: 6 hrs**

**Comment by SHN157** [comment]
### Evaluation Complete

Evaluated best model (`lr_5e4`, dev_cos=0.6269) on PHOENIX-2014-T test set (642 samples). Integrated frozen SONAR decoder to generate German text directly from predicted embeddings. Computed BLEU, BLEURT, and ROUGE-L against paper-reported numbers.

| Metric | Ours | Paper | Gap |
|--------|------|-------|-----|
| BLEU | 0.28 | 22.01 | -21.73 |
| BLEURT | 0.352 | 0.545 | -0.193 |
| ROUGE-L | 18.58 | 41.44 | -22.86 |

**Gap analysis:**
- Embedding alignment is at dev cos=0.63 — decoder receives imperfect z vectors, producing reasonable but imprecise text
- Decoder is frozen (not trained on sign language data)
- Default decoding parameters

The pipeline is verified end-to-end — the model correctly learned the weather domain and produces reasonable multilingual output (DE/EN/FR/ES). Improvement requires pushing embedding alignment higher, which is the focus of the next task).

**Deliverable:** `eval.ipynb` — evaluation script, per-metric scores, comparison table

[eval.ipynb](https://github.com/user-attachments/files/26454709/eval.ipynb)

**Comment by SHN157** [comment]
### Post-Evaluation Analysis & Next Steps

**Experiment: Increased batch size (8→32) + epochs (50→100)**
- dev_cos: 0.6269 → 0.6362
- BLEU: 0.28 → 0.34
- ROUGE-L: 18.58 → 19.67
- Best checkpoint at epoch 60

**Finding:** Scaling batch size and epochs gives marginal improvement only. 
The gap vs paper (BLEU 22.01, ROUGE-L 41.44) remains large.

**Root cause:** Frozen SONAR decoder is the bottleneck. 
The decoder was pretrained on text embeddings only — it has never seen 
sign language embeddings during training. Even at dev_cos=0.63, the 
predicted z vectors are not close enough to the text embedding space 
for the frozen decoder to generate accurate text.

**Next:** Fine-tune the decoder jointly with the student encoder so it 
learns to decode imperfect sign language embeddings directly.

**Comment by SHN157** [comment]
### Training Analysis — Plot Summary

**1. Loss Curve**
Train loss decreases smoothly. Dev loss flattens at epoch ~40 and rises 
slightly after epoch 60 — confirms overfitting. Best checkpoint correctly 
saved at epoch 60 (dev_cos=0.6362).

**2. Cosine Similarity Curve**
Both train/dev cos increase steadily. Dev cos peaks at epoch 60 (0.6362) 
then plateaus — consistent with loss curve. Train cos keeps rising after 
epoch 60 while dev cos flattens, confirming overfitting.

**3. Embedding Distribution**
z norm (0.185) is ~16% smaller than s norm (0.221) — predicted embeddings 
are consistently lower magnitude than text embeddings the decoder expects. 
This magnitude mismatch directly hurts decoder performance.

**4. Per-sample Cosine Similarity**
64.3% good (>0.6), 33.6% medium (0.4-0.6), 2% poor (<0.4).
The 33.6% medium samples are the main source of translation errors.
Mean=0.636, std=0.107 — wide spread indicates model handles some signs 
well but struggles with others.

**5. LR Schedule**
Cosine annealing decays too aggressively — LR already at 0.00018 by 
best epoch 60. 

<img width="594" height="206" alt="Image" src="https://github.com/user-attachments/assets/7178da62-3f9f-47c3-b802-063d7bafc4a7" />
<img width="573" height="254" alt="Image" src="https://github.com/user-attachments/assets/fe9d3d96-775d-4de6-ba14-fcae8842132e" />
<img width="589" height="350" alt="Image" src="https://github.com/user-attachments/assets/b0ad20dc-c890-441d-91a0-662c52099223" />
<img width="598" height="354" alt="Image" src="https://github.com/user-attachments/assets/ef4771e7-194d-43a5-8d67-d7d0c2e22780" />
